lib/services/ios-debug-service.ts

```diff
@@ -0,0 +1,294 @@
+			var emulatorDebugging = new iOSXcrunSimctl(this.$childProcess);
+			var device = emulatorDebugging.getRunningEmulator().wait();
+			if (device) {
+				this.$logger.info("Install and run on: " + device.uuid);
+			} else {
+				this.$errors.failWithoutHelp("This command expects a running emulator.");
+			}
+
+			var platformData = this.$platformsData.getPlatformData(this.platform);
+			this.$platformService.buildPlatform(this.platform).wait();
+			var emulatorPackage = this.$platformService.getLatestApplicationPackageForEmulator(platformData).wait();
+
+			device.installApp(emulatorPackage.packageName).wait();
+
+			this.executeOpenDebuggerClient().wait();
+
+			device.launchApp(this.$projectData.projectId, "--nativescript-debug-brk").wait();
+		}).future<void>()();
+	}
+
+	private emulatorStart(): IFuture<void> {
+		return (() => {
+			this.executeOpenDebuggerClient().wait();
+
+			var emulatorDebugging = new iOSXcrunSimctl(this.$childProcess);
+
+			var device = emulatorDebugging.getRunningEmulator().wait();
+			if (!device) {
+				this.$errors.failWithoutHelp("This command expects a running emulator.");
+			}
+
+			var projectId = this.$projectData.projectId;
+			var attachRequestMessage = notification.attachRequest(projectId);
+			device.postNotification(attachRequestMessage).wait();
+		}).future<void>()();
+	}
+
+	private deviceDebugBrk(): IFuture<void> {
+		return (() => {
+			this.$devicesServices.initialize({ platform: this.platform, deviceId: options.device }).wait();
+			this.$devicesServices.execute(device => (() => {
+				this.$platformService.deployOnDevice(this.platform).wait();
+				var deviceDebugging = new IOSDeviceDebugging(this.$projectData.projectId, <iOSDevice.IOSDevice>device, this.$logger, this.$injector);
+				deviceDebugging.debugApplicationOnStart();
+				this.executeOpenDebuggerClient().wait();
+				device.runApplication(this.$projectData.projectId).wait();
+			}).future<void>()()).wait();
+		}).future<void>()();
+	}
+
+	private deviceStart(): IFuture<void> {
+		return (() => {
+			this.$devicesServices.initialize({ platform: this.platform, deviceId: options.device }).wait();
+
+			this.$devicesServices.execute(device => (() => {
+				this.executeOpenDebuggerClient().wait();
+				var deviceDebugging = new IOSDeviceDebugging(this.$projectData.projectId, <iOSDevice.IOSDevice>device, this.$logger, this.$injector);
+				deviceDebugging.debugRunningApplication();
+			}).future<void>()()).wait();
+		}).future<void>()();
+	}
+
+	public executeOpenDebuggerClient(): IFuture<void> {
+		if (options.client === false) {
+			// NOTE: The --no-client has been specified. Otherwise its either false or undefined.
+			return (() => {
+				this.$logger.info("Supressing debugging client.");
+			}).future<void>()();
+		} else {
+			return this.openDebuggingClient();
+		}
+	}
+
+	private openDebuggingClient(): IFuture<void> {
+		return (() => {
+			var cmd = "open -a Safari " + this.getSafariPath();
+			this.$childProcess.exec(cmd).wait();
```

Comment:
You can remove `return (() => { .. }).future<void>()()` and just ` var cmd = "open -a Safari " + this.getSafariPath();
return this.$childProcess.exec(cmd)`